src/deck/drivers/src/clockCorrection/clockCorrectionFunctions.c

```diff
@@ -0,0 +1,97 @@
+ */
+bool emptyClockCorrectionBucket(clockCorrectionStorage_t* storage) {
+  if (storage->clockCorrectionBucket > 0) {
+    storage->clockCorrectionBucket--;
+    return false;
+  }
+
+  return true;
+}
+
+/**
+ Calculates the clock correction between a reference clock and another clock (x).
+
+ @param new_t_in_cl_reference The newest time of occurrence for an event (t), meassured by the clock of reference
+ @param old_t_in_cl_reference An older time of occurrence for an event (t), meassured by the clock of reference
+ @param new_t_in_cl_x The newest time of occurrence for an event (t), meassured by clock x
+ @param old_t_in_cl_x The previous time of occurrence for an event (t), meassured by clock x
+ @return The necessary clock correction to apply to timestamps meassured by clock x, in order to obtain their value like if the meassurement was done by the reference clock. Or -1 if it was not possible to perform the computation. Example: timestamp_in_cl_reference = clockCorrection * timestamp_in_cl_x
+ */
+double calculateClockCorrection(const uint64_t new_t_in_cl_reference, const uint64_t old_t_in_cl_reference, const uint64_t new_t_in_cl_x, const uint64_t old_t_in_cl_x) {
+  const uint64_t tickCount_in_cl_reference = truncateTimeStampFromDW1000(new_t_in_cl_reference - old_t_in_cl_reference);
+  const uint64_t tickCount_in_cl_x = truncateTimeStampFromDW1000(new_t_in_cl_x - old_t_in_cl_x);
+
+  if (tickCount_in_cl_x == 0) {
+    return -1;
+  }
+
+  return (double)tickCount_in_cl_reference / (double)tickCount_in_cl_x;
+}
+
+/**
+ Updates the clock correction only if the provided value follows certain conditions. This is used to discard wrong clock correction meassurements.
+ @return True if the provided clock correction was valid and accepted, false otherwise.
+ */
+bool updateClockCorrection(clockCorrectionStorage_t* storage, const double clockCorrectionCandidate) {
+  bool sampleIsAccepted = false;
+
+  if (CLOCK_CORRECTION_SPEC_MIN < clockCorrectionCandidate && clockCorrectionCandidate < CLOCK_CORRECTION_SPEC_MAX) {
+    const double currentClockCorrection = storage->clockCorrection;
+    const double difference = clockCorrectionCandidate - currentClockCorrection;
+
+    if (-CLOCK_CORRECTION_ACCEPTED_NOISE < difference && difference < CLOCK_CORRECTION_ACCEPTED_NOISE) {
+      // Simple low pass filter
+      const double newClockCorrection = currentClockCorrection * CLOCK_CORRECTION_FILTER + clockCorrectionCandidate * (1.0 - CLOCK_CORRECTION_FILTER);
+
+      fillClockCorrectionBucket(storage);
+
+      storage->clockCorrection = newClockCorrection;
+      sampleIsAccepted = true;
+    } else {
+      const bool shouldAcceptANewClockReference = emptyClockCorrectionBucket(storage);
+      if (shouldAcceptANewClockReference) {
+        fillClockCorrectionBucket(storage);
+        storage->clockCorrection = clockCorrectionCandidate;
+        sampleIsAccepted = true;
```

Comment:
In previous implementation, when the sample was accepted after the bucket was emptied, `fillClockCorrectionBucket(storage)` and `sampleIsAccepted = true`where not called. As a result, the clock was set, but the function was returning false.